docs/app/Examples/modules/Accordion/Advanced/AccordionExampleForm.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { Accordion, Button, Form, Segment } from 'semantic-ui-react'
+
+const panels = [
+  {
+    title: 'Optional Details',
+    content: {
+      as: Form.Input,
+      key: 'content',
+      label: 'Maiden Name',
+      placeholder: 'Maiden Name',
+    },
+  },
+]
+
+const AccordionExampleForm = () => (
+  <Segment>
+    <Form>
+      <Form.Group widths={2}>
+        <Form.Input label='First Name' placeholder='First Name' />
+        <Form.Input label='First Name' placeholder='Last Name' />
+      </Form.Group>
+      <Accordion as={Form.Field} panels={panels} />
+
+      <Button secondary>Sign Up</Button>
+    </Form>
+  </Segment>
+)
```

Comment:
Not sure about the API here, this produces invalid SUI HTML with two nested fields:

```html
<div class="field accordion ui">
  <div class="title">
    <i class="dropdown icon"></i>Optional Details
  </div>
  <div class="field content">
    <label>Maiden Name</label>
    <div class="ui input">
      <input type="text" placeholder="Maiden Name" />
    </div>
  </div>
</div>
```

The `Form.Field` creates a field, but the content as `Form.Input` also creates a field.  I tried to correct this by making the Accordion render `as` a `Form.Field` with a `label` prop, and the content render `as` an `Input` with the `placeholder`, however, this does not work.

I think I see where this is trying to go by allowing any component to be the Accordion and any component to be the Content, however, the CSS doesn't allow for this.  We're bound to hit issues merging them.

The `ui accordion` should only contain children, same as the `title` and `content`.  It would be ideal to have a generic Accordion component that could take any component, however, I think it would have to be JS style based and not CSS className based.

Reply:
I will check this example tomorrow. @levithomason do you have another remarks to this PR? After it will be merged, it will be easiest place where we can get transitions :smile:

Reply:
Hm, @levithomason it renders same as [there](https://semantic-ui.com/modules/accordion.html#form-fields) except CSS classes order. Can you take a look? Did I miss something?

```html
    <div class="ui accordion field">
      <div class="title">
        <i class="icon dropdown"></i>
        Optional Details
      </div>
      <div class="content field">
        <label>Maiden Name</label>
        <input placeholder="Maiden Name" type="text">
      </div>
    </div>
```

Reply:
I'm saying I do not think the HTML is valid SUI markup.  I see no supported examples of nested fields: `field > field`.

Reply:
Take a look [there](https://semantic-ui.com/modules/accordion.html#form-fields)

```html
<div class="ui accordion field">
  <div class="content field active"></div>
</div>
```

Reply:
Thanks for straightening me out on that!  Will manually test this to refresh my memory.  Likely will merge.